### DISCUSSION TITLE
Read-only collections based on SQLite views

### DISCUSSION THREAD

**Discussion by daniel-zlatev** [pocketbase/pocketbase#212] [Q&A] [answered]
Hi there, I think it will be quite usefull to have the flexibility for creating collections based on SQLite views. 

The benefits:
- Full access to the native SQLite functions - Math, Text, etc
- Increased flexibility for filtering using view columns (time based filtering will be a breeze) 
- Aggregations, Window functions and Joins will be trivial to implement

The downsides:
- Those collections will be read-only, due the nature of the sql views. This may add unexpected complexy to the codebase.

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure that I understand the feature request.

How do you intend to use a collection which underlying table is read-only? Just for filtering and aggregations?
If that's the case, for now, I'm not planning adding support for aggregations (or reporting in general) out of the box because usually such functions operate over multiple datasets, sometimes combine even several fields into one and it is hard to generalize.

If you need aggregations or any more advanced query capabilities, the recommended approach for now is to use PocketBase as framework and directly to access the internal db abstraction. You could find some documentation on this topic here - https://pocketbase.io/docs/db-queries/ (more advanced examples and guides will be added in the near future).
